include/extractor/serialization.hpp

```diff
@@ -81,6 +85,54 @@ inline void write(storage::io::FileWriter &writer,
     storage::serialization::write(writer, turn_data_container.pre_turn_bearings);
     storage::serialization::write(writer, turn_data_container.post_turn_bearings);
 }
+
+// read/write for conditional turn restrictions file
+inline void read(const boost::filesystem::path &path, std::vector<TurnRestriction> &restrictions)
+{
+    const auto fingerprint = storage::io::FileReader::VerifyFingerprint;
+    storage::io::FileReader reader{path, fingerprint};
+
+    auto num_indices = reader.ReadElementCount64();
+    restrictions.resize(num_indices);
+    TurnRestriction restriction;
+    while (num_indices > 0)
+    {
+        bool is_only;
+        reader.ReadInto(restriction.via);
+        reader.ReadInto(restriction.from);
+        reader.ReadInto(restriction.to);
+        reader.ReadInto(is_only);
+        auto num_conditions = reader.ReadElementCount64();
+        restriction.condition.resize(num_conditions);
+        for (uint64_t i = 0; i < num_conditions; i++)
+        {
+            reader.ReadInto(restriction.condition[i].modifier);
+            storage::serialization::read(reader, restriction.condition[i].times);
+            storage::serialization::read(reader, restriction.condition[i].weekdays);
+            storage::serialization::read(reader, restriction.condition[i].monthdays);
+        }
+        restriction.flags.is_only = is_only;
+
+        restrictions.push_back(restriction);
```

Comment:
Arghhh sorry comments like these are very helpful :pray: You already explained this to me once though, I was using an `std::move` before and then I must have rebased it away. Thanks for pointing out again!